pkg/network/aci_network.go

```diff
@@ -128,23 +128,26 @@ func (pn *ProviderNetwork) setupNetwork(ctx context.Context, azConfig *auth.Conf
 	var rawResponse *http.Response
 	ctxWithResp := runtime.WithCaptureResponse(ctx, &rawResponse)
 
-	createSubnet := true
+	updateSubnet := true
 	currentSubnet, err := pn.GetACISubnet(ctxWithResp, subnetsClient)
 	if err != nil {
 		return err
 	}
 
-	if err == nil {
-		createSubnet, err = pn.shouldCreateSubnet(currentSubnet, createSubnet)
+	createNewSubnet := false
```

Comment:
The code logic is correct. But can you slightly improve the code readability?

You can initialize updateSubnet to false.

line 148, change it to

if createNewSubnet || updateSubnet: